pkg/scalers/arangodb_scaler.go

```diff
@@ -0,0 +1,276 @@
+	// +optional
+	connectionLimit int64
+
+	// The index of the scaler inside the ScaledObject
+	// +internal
+	scalerIndex int
+}
+
+// NewArangoDBScaler creates a new arangodbScaler
+func NewArangoDBScaler(config *ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
+	}
+
+	meta, err := parseArangoDBMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing arangoDB metadata: %s", err)
+	}
+
+	client, err := getNewArangoDBClient(meta)
+	if err != nil {
+		return nil, err
+	}
+
+	return &arangoDBScaler{
+		metricType: metricType,
+		metadata:   meta,
+		client:     client,
+		logger:     InitializeLogger(config, "arangodb_scaler"),
+	}, nil
+}
+
+func getNewArangoDBClient(meta *arangoDBMetadata) (driver.Client, error) {
+	var auth driver.Authentication
+
+	conn, err := http.NewConnection(http.ConnectionConfig{
+		Endpoints: strings.Split(meta.endpoints, ","),
+		TLSConfig: &tls.Config{
+			MinVersion:         tls.VersionTLS13,
+			InsecureSkipVerify: meta.unsafeSsl,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a new http connection, %v", err)
+	}
+
+	if meta.arangoDBAuth.EnableBasicAuth {
+		auth = driver.BasicAuthentication(meta.arangoDBAuth.Username, meta.arangoDBAuth.Password)
+	} else if meta.arangoDBAuth.EnableBearerAuth {
+		hdr, err := jwt.CreateArangodJwtAuthorizationHeader(meta.arangoDBAuth.BearerToken, meta.serverID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create bearer token authorization header, %v", err)
+		}
+		auth = driver.RawAuthentication(hdr)
+	}
+
+	client, err := driver.NewClient(driver.ClientConfig{
+		Connection:     conn,
+		Authentication: auth,
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize a new client, %v", err)
+	}
+
+	return client, nil
+}
+
+func parseArangoDBMetadata(config *ScalerConfig) (*arangoDBMetadata, error) {
+	var err error
+
+	// setting default metadata
+	meta := arangoDBMetadata{}
+
+	// parse metaData from ScaledJob config
+	meta.endpoints, err = GetFromAuthOrMeta(config, "endpoints")
+	if err != nil {
+		return nil, err
+	}
+
+	if val, ok := config.TriggerMetadata["collection"]; ok {
+		meta.collection = val
+	} else {
+		return nil, fmt.Errorf("no collection given")
+	}
+
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert queryValue to int, %v", err)
+		}
+		meta.queryValue = queryValue
+	} else {
+		return nil, fmt.Errorf("no queryValue given")
+	}
+
+	meta.activationQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
+		activationQueryValue, err := strconv.ParseInt(val, 10, 64)
```

Comment:
same as above

Reply:
We're counting the number of records for a given query, similar to MongoDB, hence it would always be an integer.